libs/liblobby/lobby/interface.lua

```diff
@@ -648,7 +648,128 @@ Interface.commands["FRIENDREQUESTLISTEND"] = Interface._OnFriendRequestListEnd
 -- Battle commands
 ------------------------
 
+
+-- if num1 > num2 return 1
+-- if num1 < num2 return 0
+-- if num1 = num2 return -1
+local function bignumStr_isGreater(num1, num2)
+	if #num1 > #num2 then
+		return 1
+	end
+	if #num1 < #num2 then
+		return 0
+	end
+  
+	local dig1, dig2
+	for i=1, #num1,1 do
+		dig1 = tonumber(string.sub(num1, i, i))
+		dig2 = tonumber(string.sub(num2, i, i))
+		if dig1 > dig2 then
+			return 1
+		elseif dig1 < dig2 then
+			return 0
+		end
+	end
+	-- equal
+	return -1
+end
+
+-- Subtraction for numbers given in string format greater than 24bits numbers (Spring Lua numbers are floats with maximum of 24bit)
+local function bignumStr_subtraction(minuid, subtrahend)
```

Comment:
TBH I don't exactly understand the logic here, you are using a bunch of maps, IMHO this can be much simpler:


```lua
-- works only for positive integers!
local function bignum_substract(num1, num2)
	local c = bignum_compare(num1, num2)
	sign = ""
	if c == 0 then
		return "0"
	elseif c == -1 then
		sign = "-"
		num1, num2 = num2, num1
	end
	while #num2 < #num1 do
		num2 = "0" .. num2
	end

	-- actual substraction
	local borrow = 0
	local res = ""
	for i=#num1, 1, -1 do
		local d1 = tonumber(num1:sub(i, i)) - borrow
		local d2 = tonumber(num2:sub(i, i))
		borrow = 0
		if d1 < d2 then
			d1 = d1 + 10
			borrow = 1
		end
		res = res .. (d1 - d2)
	end
	res = res:reverse()

	-- strip leading 0s
	while res:sub(1, 1) == "0" do
		res = res:sub(2, #res)
	end
	return sign .. res
end
```